lib/evaluate/type.cc

```diff
@@ -150,14 +150,11 @@ bool DynamicType::IsTypeCompatibleWith(const DynamicType &that) const {
 // corresponding kind type parameters of the type2?
 static bool IsKindCompatible(const semantics::DerivedTypeSpec &type1,
     const semantics::DerivedTypeSpec &type2) {
-  for (const auto &[name, symbol] : *type1.scope()) {
-    if (const auto *details{symbol->detailsIf<semantics::TypeParamDetails>()}) {
-      if (details->attr() == common::TypeParamAttr::Kind) {
-        const semantics::ParamValue *param1{type1.FindParameter(name)};
-        const semantics::ParamValue *param2{type2.FindParameter(name)};
-        if (!PointeeComparison(param1, param2)) {
-          return false;
-        }
+  for (const auto &[name, param1] : type1.parameters()) {
```

Comment:
Please ensure that both PDTs have had their defaulted type parameters made explicit in their `ParamValue` lists.  I think that they do, but the representation has changed a few times during development.

Reply:
Looking at the code, I think it is happening here:
https://github.com/flang-compiler/f18/blob/46d77ab006fd5a633ea9ab18b208ac9a7baee4f2/lib/semantics/tools.cc#L695
I have added tests that involved implicit kind parameters to ensure this.